Melt/identify_species.py

```diff
@@ -0,0 +1,137 @@
+
+import librosa
+import numpy as np
+import os
+import tensorflow as tf
+print(tf.__version__)
+
+frequency_min = 600
+frequency_max = 1200
+num_bands = int((frequency_max - frequency_min) / 10)
+slices_per_second = 20
+seconds_per_sample = 3.0
+slices_per_sample = int(slices_per_second * seconds_per_sample)
+sample_slide_seconds = 1.0
+sample_slide_slices = int(sample_slide_seconds * slices_per_second)
+accept_threshold = 0.5
+
+model_file_name = 'saved_model.pb'
+
+
+def _load_sample(path):
+    frames, sr = librosa.load(path, sr=None)
+
+    # generate spectrogram
+    nfft = int(sr / 10)
+    stft = librosa.stft(frames, n_fft=nfft, hop_length=int(nfft / 2))
+    npspec = np.abs(stft)[int(frequency_min / 10):int(frequency_max / 10)]
+
+    return sr, npspec
+
+def _model_paths(basepath):
+    namelist = os.listdir(basepath)
+    pathlist = list()
+    for name in namelist:
+        namepath = os.path.join(basepath, name)
+        if os.path.isdir(namepath):
+            pathlist = pathlist + _model_paths(namepath)
+        elif namepath.endswith(model_file_name):
+            pathlist.append(basepath)
+    return pathlist
+
+def _find_likely_span(liklihoods, first, last):
```

Comment:
not really sure what is going on here, but looks like it's always returning  `max(liklihoods[first:last + 1])`  so just calculate that once. is count  the number of samples?

It looks to me like all of these conditions could be condensed and written generallically using similar logic to line:69

Reply:
Is this returning the span in seconds? to make this clearer, the parameters could be `sample_start`, `sample_end` and it returns `start_s`, `end_s` or something like that and a bit of a description about what this function is actually doing

Reply:
rewrote this to document and clarify what's going on